scripts/annotations.py

```diff
@@ -0,0 +1,99 @@
+"""
+Author: Alexandra Lee
+Date Created: 20 September 2021
+
+Scripts to annotate genes
+"""
+import pandas as pd
+from . import utils
+
+
+def load_format_operons(operon_filename):
+    """
+    This function inputs operon annotations from the `operon_filename`
+    and returns a pandas df with gene ids as index and the associated
+    operon as the column values.
+    """
+
+    operon_df = pd.read_csv(operon_filename, index_col=0, header=0)
+
+    # There are 247 PAO1 genes with multiple annotations
+    # This operon df contains annotations from predicted operons based on DOOR database
+    # predictions which make up the majority of the operons) as well as some that
+    # are curated (i.e. PseudoCAP)
+    # There are some that have multiple PseudoCAP annotations too
+
+    # Here we will keep the last PseudoCAP annotations
+    # To ensure that the PseudoCAP annotations are the last ones, we will sort the values
+    operon_df = operon_df.sort_values(by=["locus_tag", "source_database"])
+    operon_df = operon_df.set_index("locus_tag")
+    operon_df = operon_df[~operon_df.index.duplicated(keep="last")]
```

Comment:
This works, but you can make your life easier in the future with [drop_duplicates](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.drop_duplicates.html).